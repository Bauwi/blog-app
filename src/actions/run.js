import database from '../firebase/firebase';
import { db } from '../firebase/firebase';
import * as firebase from 'firebase';

// manage CRUD of run
export const addPostToRun = post => ({
  type: 'ADD_POST_TO_RUN',
  post
});

export const startAddPostToRun = post => (dispatch, getState) => {
  const { uid } = getState().auth;
  db
    .collection('runs')
    .doc(uid)
    .collection('UserRuns')
    .add(post)
    .then((doc) => {
      dispatch(addPostToRun({
        DBid: doc.id,
        ...post
      }));
    });
};

export const removePostToRun = id => ({
  type: 'REMOVE_POST_TO_RUN',
  id
});

export const startRemovePostToRun = (id, DBid) => (dispatch, getState) => {
  const { uid } = getState().auth;
  return db
    .collection('runs')
    .doc(uid)
    .collection('UserRuns')
    .doc(`${DBid}`)
    .delete()
    .then(dispatch(removePostToRun(id)));
};

// set Run posts. This action is dispatched in the header component
// as it needs to be available for the run dropdown

export const runsHasErrored = bool => ({
  type: 'RUN_HAS_ERRORED',
  hasErrored: bool
});

export const runIsLoading = bool => ({
  type: 'RUN_IS_LOADING',
  isLoading: bool
});

export const setRunPosts = posts => ({
  type: 'SET_RUN_POSTS',
  posts
});

export const startSetRunPosts = resume => (dispatch, getState) => {
  dispatch(runIsLoading(true));
  const { uid } = getState().auth;
  const run = [];
  return db
    .collection('runs')
    .doc(uid)
    .collection('UserRuns')
    .get()
    .then((querySnapshot) => {
      querySnapshot.forEach((doc) => {
        run.push({
          DBid: doc.id,
          ...doc.data()
        });
      });
      return run;
    })
    .then(run => dispatch(setRunPosts(run)))
    .then(() => {
      const firstUnreadIndex = run.findIndex(post => post.state === 'unread');
      if (firstUnreadIndex === -1) {
        return dispatch(setCurrentPostRun(run[0].content.id));
      }
      return dispatch(setCurrentPostRun(run[firstUnreadIndex].content.id));
    })
    .then(() => dispatch(runIsLoading(false)))
    .catch(() => dispatch(runsHasErrored(true)));
};

// change the status of the past from read to unread
export const updateRunPostToAlreadyRead = id => ({
  type: 'UPDATE_RUN_POST_TO_ALREADY_READ',
  id
});

export const startUpdateRunPostToAlreadyRead = (id, DBid) => (dispatch, getState) => {
  const { uid } = getState().auth;
  return db
    .collection('runs')
    .doc(uid)
    .collection('UserRuns')
    .doc(`${DBid}`)
    .update({ state: 'read' })
    .then(() => {
      dispatch(updateRunPostToAlreadyRead(id));
    });
};

// this one is not used anymore. But it could be used to mark some popst as unread yet
export const updateRunPostToNotAlreadyRead = id => ({
  type: 'UPDATE_RUN_POST_TO_NOT_ALREADY_READ',
  id
});

// Set which post should actually be current. Run navigation depends on this.
// instead of routing, component is updated with currentPost data.
// easier way to deal with "Resume" to provide the user the first unread post in its run.
export const setCurrentPostRun = id => ({
  type: 'SET_CURRENT_POST_RUN',
  id
});

export const startSetCurrentPostRun = id => (dispatch) => {
  dispatch(runIsLoading(true));
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve();
    }, 10);
  })
    .then(() => dispatch(setCurrentPostRun(id)))
    .then(() => dispatch(runIsLoading(false)))
    .catch(() => dispatch(runsHasErrored(true)));
};

export const resetRun = () => ({
  type: 'RESET_RUN'
});

// remove all posts in tun
export const startResetRun = () => (dispatch, getState) => {
  const { uid } = getState().auth;
  return db
    .collection('runs')
    .doc(uid)
    .collection('UserRuns')
    .get()
    .then((querySnapshot) => {
      querySnapshot.forEach((doc) => {
        db
          .collection('runs')
          .doc(uid)
          .collection('UserRuns')
          .doc(doc.ref.id)
          .delete();
      });
    })
    .then(() => dispatch(resetRun()));
};

// Remove already read posts in run
export const cleanRun = () => ({
  type: 'CLEAN_RUN'
});

export const startCleanRun = alreadyRead => (dispatch, getState) => {
  const { uid } = getState().auth;
  return db
    .collection('runs')
    .doc(uid)
    .collection('UserRuns')
    .get()
    .then((querySnapShot) => {
      querySnapShot.forEach((doc) => {
        if (alreadyRead.includes(doc.ref.id)) {
          db
            .collection('runs')
            .doc(uid)
            .collection('UserRuns')
            .doc(doc.ref.id)
            .delete();
        }
      });
    })
    .then(dispatch(cleanRun()));
};
