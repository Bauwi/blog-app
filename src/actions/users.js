import { db } from '../firebase/firebase';

export const updateUser = (id, updates) => ({
  type: 'UPDATE_USER',
  id,
  updates
});

export const startUpdateUser = (id, updates) => (dispatch, getState) => {
  const uid = id || getState().auth.uid;
  return db
    .collection('users')
    .doc(uid)
    .get()
    .then((doc) => {
      if (doc.exists) {
        return db
          .collection('users')
          .doc(uid)
          .update(updates)
          .then(() => {
            dispatch(updateUser(uid, updates));
          });
      }
      return db
        .collection('users')
        .doc(uid)
        .set({
          avatar: 'http://www.perlworkshop.nl/nlpw2016/img/default-avatar.png',
          email: '',
          username: '',
          stars: 0,
          topCategories: ['life', 'programming', 'Music']
        })
        .then(startUpdateUser(uid, updates));
    });
};

export const addUserStar = nextStars => ({
  type: 'ADD_USER_STAR',
  nextStars
});

export const startAddUserStar = (id, prevStars) => dispatch =>
  db
    .collection('users')
    .doc(id)
    .update({ stars: prevStars + 1 })
    .then(() => dispatch(addUserStar(prevStars + 1)));

export const setAuthor = author => ({
  type: 'SET_AUTHOR',
  author
});

export const startSetAuthor = id => dispatch =>
  db
    .collection('users')
    .doc(id)
    .get()
    .then((doc) => {
      dispatch(setAuthor(doc.data()));
    });

export const setUserPreferences = preferences => ({
  type: 'SET_USER_PREFERENCES',
  preferences
});

export const startSetUserPreferences = preferences => (dispatch, getState) =>
  db
    .collection('users')
    .doc(getState().auth.uid)
    .get()
    .then((doc) => {
      dispatch(setUserPreferences(doc.data()));
    });
