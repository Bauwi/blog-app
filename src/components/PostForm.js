import React, { Component } from 'react';
import moment from 'moment';

export default class PostForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      title: props.post ? props.post.title : '',
      body: props.post ? props.post.body : '',
      error: '' 
    };
  }
  
  onTitleChange = (e) => {
    const title = e.target.value;
    this.setState(() => ({ title }));
  }

  onBodyChange = (e) => {
    const body= e.target.value;
    this.setState(() => ({ body }));
  }

  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.title || !this.state.body) {
      this.setState(() => ({ error: 'Please provide a title and body to your post.' }))
    } else {
      this.setState(() => ({ error: '' }));
      this.props.onSubmit({
        title: this.state.title,
        body: this.state.body
      });
    }
  }; 

  render() {
    return (
      <form onSubmit={this.onSubmit}>
        <input
          autoFocus
          placeholder="title"
          type="text"
          value={this.state.title}
          onChange={this.onTitleChange}   
        />
        {this.state.title}
        <textarea 
          placeholder="Add a post to your blog."
          value={this.state.body}
          onChange={this.onBodyChange}
        >
        </textarea>
        {this.state.body}
        <button>Save Post</button>
      </form>
    )
  }
}