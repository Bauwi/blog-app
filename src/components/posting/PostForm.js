import React, { Component } from 'react';
import moment from 'moment';
import ReactQuill from 'react-quill';

export default class PostForm extends Component {
  constructor(props) {
    super(props);
    // bodyToEditor is a tweak as db does not handle delta format from quill Editor
    // body is the part sent to db and bodyToEditor the data used as the correct format
    // to display content ==> delta standard format: delta {ops: [array of data]}
    this.state = {
      title: props.post ? props.post.title : '',
      body: props.post ? props.post.body : '',
      bodyToEditor: props.post ? { ops: [...this.props.post.body] } : '',
      shortBody: props.post ? props.post.shortBody : '',
      keywords: props.post ? props.post.keywords : '',
      createdAt: props.post ? moment(props.post.createdAt) : moment(),
      author: props.post ? props.post.author : '',
      readingTime: props.post ? props.post.readingTime : 0,
      cover: props.post ? props.post.cover : '',
      error: ''
    };
  }

  onTitleChange = e => {
    const title = e.target.value;
    this.setState(() => ({ title }));
  };

  onBodyChange = body => {
    if (this.refs.quill) {
      const body = this.refs.quill.editor.editor.delta;
      const bodyToEditor = { ops: [...body] };

      const shortBody =
        this.refs.quill.editor.getText().length > 250
          ? this.refs.quill.editor
              .getText()
              .slice(0, 250)
              .concat('...')
          : this.refs.quill.editor.getText();
      const readingTime = this.readingTime(this.refs.quill.editor.getText());

      this.setState(() => ({
        body: body.ops,
        shortBody,
        bodyToEditor,
        readingTime
      }));
    }
  };

  onAuthorChange = e => {
    const author = e.target.value;
    this.setState(() => ({ author }));
  };

  onKeywordsChange = e => {
    const keywords = e.target.value;
    this.setState(() => ({ keywords }));
  };
  onCoverChange = e => {
    const cover = e.target.value;
    this.setState(() => ({ cover }));
  };

  readingTime = str => {
    return Math.floor(str.split(' ').length / 250);
  };

  onSubmit = e => {
    e.preventDefault();
    if (!this.state.title || !this.state.body) {
      this.setState(() => ({ error: 'Please provide a title and body to your post.' }));
    } else {
      this.setState(() => ({ error: '' }));
      this.props.onSubmit({
        title: this.state.title,
        body: this.state.body,
        shortBody: this.state.shortBody,
        keywords: this.state.keywords,
        createdAt: this.state.createdAt.valueOf(),
        author: this.state.author,
        readingTime: this.state.readingTime,
        cover: this.state.cover
      });
    }
  };

  render() {
    return (
      <form className="content-container" onSubmit={this.onSubmit}>
        <input
          autoFocus
          placeholder="title"
          type="text"
          value={this.state.title}
          onChange={this.onTitleChange}
        />
        {this.state.title}
        <input
          type="text"
          placeholder="author"
          value={this.state.author}
          onChange={this.onAuthorChange}
        />
        <input
          type="text"
          placeholder="keywords"
          value={this.state.keywords}
          onChange={this.onKeywordsChange}
        />
        <input
          type="text"
          placeholder="cover"
          value={this.state.cover}
          onChange={this.onCoverChange}
        />
        <div>
          <ReactQuill
            ref="quill"
            theme="snow"
            defaultValue={this.state.bodyToEditor}
            modules={PostForm.modules}
            formats={PostForm.formats}
            bounds={'.app'}
            placeholder="Add a post to your blog"
            onChange={this.onBodyChange}
          />
        </div>

        <button>Save Post</button>
      </form>
    );
  }
}

PostForm.modules = {
  toolbar: [
    [{ header: '1' }, { header: '2' }],

    ['bold', 'italic', 'underline', 'blockquote', 'code-block'],
    [{ size: ['small', false, 'large', 'huge'] }],

    [{ color: [] }, { background: [] }],

    [{ list: 'ordered' }],
    ['link', 'image', 'video'],
    [{ align: [] }],
    ['clean']
  ]
};

PostForm.formats = [
  'header',
  'bold',
  'size',
  'color',
  'background',
  'italic',
  'underline',
  'blockquote',
  'code-block',
  'list',
  'link',
  'image',
  'video',
  'align'
];
