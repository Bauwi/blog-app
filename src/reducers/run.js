const runReducerDefaultState = {
  isLoading: true, posts: [], hasErrored: false, current: {}
};

const runReducer = (state = runReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_POST_TO_RUN':
      return {
        ...state,
        posts: [...state.posts, action.post]
      };
    case 'REMOVE_POST_TO_RUN':
      return {
        ...state,
        posts: state.posts.filter(post => action.id !== post.content.id)
      };
    case 'SET_RUN_POSTS':
      return {
        ...state,
        posts: action.posts
      };
    case 'RUN_IS_LOADING':
      return {
        ...state,
        isLoading: action.isLoading
      };
    case 'RUN_HAS_ERRORED':
      return {
        ...state,
        hasErrored: action.hasErrored
      };
    case 'SET_CURRENT_POST_RUN':
      return {
        ...state,
        current: state.posts.find(post => post.content.id === action.id)
      };
    case 'UPDATE_RUN_POST_TO_ALREADY_READ':
      return {
        ...state,
        current: { ...state.current, state: 'read' },
        posts: state.posts.map((post) => {
          if (post.content.id === action.id) {
            return {
              ...post,
              state: 'read'
            };
          }
          return post;
        })
      };
    case 'UPDATE_RUN_POST_TO_NOT_ALREADY_READ':
      return {
        ...state,
        posts: state.posts.map((post) => {
          if (post.content.id === action.id) {
            return {
              ...post,
              state: 'unread'
            };
          }
          return post;
        })
      };
    case 'RESET_RUN':
      return {
        posts: []
      };
    case 'CLEAN_RUN':
      return {
        ...state,
        posts: state.posts.filter(post => post.state === 'unread')
      };
    default:
      return state;
  }
};

export default runReducer;
