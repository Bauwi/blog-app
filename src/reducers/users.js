const usersReducerDefaultState = [];

const usersReducer = (state = usersReducerDefaultState, action) => {
  switch (action.type) {
    case 'CREATE_USER':
      return {
        ...state,
        preferences: action.newUser
      };
    case 'UPDATE_USER_PREFERENCES':
      return {
        ...state,
        preferences: { ...state.preferences, ...action.updates }
      };
    case 'SET_AUTHOR':
      return {
        ...state,
        author: action.author
      };
    case 'ADD_USER_STAR':
      return {
        ...state,
        author: {
          ...state.author,
          stars: action.nextStars
        }
      };
    case 'SET_USER_PREFERENCES':
      return {
        ...state,
        preferences: action.preferences
      };
    case 'USERS_IS_LOADING':
      return {
        ...state,
        isLoading: action.isLoading
      };
    case 'USERS_HAS_ERRORED':
      return {
        ...state,
        hasErrored: action.hasErrored
      };
    default:
      return state;
  }
};

export default usersReducer;
