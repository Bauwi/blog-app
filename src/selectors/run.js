const getRunPosts = (posts = [], run) => run.map(read => posts.find(post => post.id === read.id));
export default getRunPosts;
