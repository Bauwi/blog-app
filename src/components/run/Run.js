import React, { Component } from 'react';
import { connect } from 'react-redux';

import InlineList from '../InlineList';
import LoadingPage from '../LoadingPage';
import { startSetRunPosts } from '../../actions/run';

export class Run extends Component {
  state = {
    loading: true
  };
  componentWillMount() {
    if (this.props.run.length === 0) {
      this.props.startSetRunPosts().then(() => {
        this.setState(() => ({ loading: false }));
      });
    } else {
      this.setState(() => ({ loading: false }));
    }
  }

  render() {
    if (this.state.loading) {
      return <LoadingPage />;
    }
    return (
      <div>
        <p>Run</p>
        <InlineList posts={this.props.run} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startSetRunPosts: () => dispatch(startSetRunPosts())
});

const mapStateToProps = state => ({
  run: state.run
});

export default connect(mapStateToProps, mapDispatchToProps)(Run);
