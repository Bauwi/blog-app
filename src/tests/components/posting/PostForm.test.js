import React from 'react';
import { shallow, mount } from 'enzyme';
import moment from 'moment';
import PostForm from '../../../components/posting/PostForm';
import { user2 } from '../../fixtures/users';
import run from '../../fixtures/run';

test('should render PostForm Component properly', () => {
  const wrapper = shallow(<PostForm author={user2} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render PostForm with correct data in', () => {
  const wrapper = shallow(<PostForm author={user2} post={run.posts[0].content} />);
  expect(wrapper).toMatchSnapshot();
});

test('should display an error message if invalid form submission', () => {
  const wrapper = shallow(<PostForm author={user2} />);
  wrapper.find('form').simulate('submit', { preventDefault: () => {} });
  expect(wrapper.state('error')).toBe(true);
});

test('should change keywords on keywords input change', () => {
  const value = 'banana, apple, orange';
  const wrapper = shallow(<PostForm author={user2} />);
  wrapper
    .find('input')
    .at(0)
    .simulate('change', {
      target: { value }
    });
  expect(wrapper.state().keywords).toBe(value);
});

test('should change title on title input change', () => {
  const value = 'new title';
  const wrapper = shallow(<PostForm author={user2} />);
  wrapper.find('TextArea').simulate('change', { target: { value } });
  expect(wrapper.state().title).toBe(value);
});

test('should call onSubmit prop with valid form submission', () => {
  const onSubmitSpy = jest.fn();
  const wrapper = shallow(<PostForm post={run.posts[0].content} author={user2} onSubmit={onSubmitSpy} />);
  wrapper.find('form').simulate('submit', {
    preventDefault: () => {}
  });
  expect(wrapper.state().error).toBe(false);
  expect(onSubmitSpy).toHaveBeenLastCalledWith({
    author: 'user2',
    cover: 'https://www.w3schools.com/w3css/img_fjords.jpg',
    miniCover: 'https://www.w3schools.com/w3css/img_fjords.jpg',
    title: 'Post 1 in Run',
    body: [{ text: 'This is the first Post in run', attributes: 'h1' }],
    keywords: 'red, red, green, blue',
    readingTime: 7,
    createdAt: 0
  });
});

// test('should change body on title', () => {
//   const value = { ops: [{ text: 'hello world !', attributes: 'h1' }] };
//   const wrapper = shallow(<PostForm author={user2} post={run.posts[1].content} />);
//   wrapper.find('Quill').simulate('change', { target });
//   expect(wrapper.state().bodyToEditor).toBe(value);
// });
