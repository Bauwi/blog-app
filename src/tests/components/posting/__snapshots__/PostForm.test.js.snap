// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PostForm Component properly 1`] = `
<form
  className="content-container form form--post"
  onSubmit={[Function]}
>
  <UserCard
    author={
      Object {
        "avatar": "http://www.perlworkshop.nl/nlpw2016/img/default-avatar.png",
        "description": "User2 description",
        "email": "",
        "numberOfPosts": 7,
        "stars": 20,
        "topCategories": Array [
          "life",
          "business",
          "history",
        ],
        "username": "user2",
      }
    }
    post={
      Object {
        "author": "user2",
        "authorId": "todo",
        "createdAt": "1970-01-01T00:00:00.000Z",
        "readingTime": 0,
      }
    }
  />
   
  <label
    htmlFor="title"
  >
    <p>
       Title: 
    </p>
     
    <p>
       What should stand for title ? 
    </p>
    <TextArea
      autosize={
        Object {
          "maxRows": 6,
          "minRows": 2,
        }
      }
      onChange={[Function]}
      placeholder="As if you could kill time without injuring eternity."
      prefixCls="ant-input"
      type="text"
      value=""
    />
     
  </label>
   
  <div>
    <Quill
      bounds=".app"
      defaultValue={
        Object {
          "ops": Array [],
        }
      }
      formats={
        Array [
          "header",
          "bold",
          "size",
          "color",
          "background",
          "italic",
          "underline",
          "blockquote",
          "code-block",
          "list",
          "link",
          "image",
          "video",
          "align",
        ]
      }
      modules={
        Object {
          "toolbar": Array [
            Array [
              Object {
                "header": "1",
              },
              Object {
                "header": "2",
              },
            ],
            Array [
              "bold",
              "italic",
              "underline",
              "blockquote",
              "code-block",
            ],
            Array [
              Object {
                "size": Array [
                  "small",
                  false,
                  "large",
                  "huge",
                ],
              },
            ],
            Array [
              Object {
                "color": Array [],
              },
              Object {
                "background": Array [],
              },
            ],
            Array [
              Object {
                "list": "ordered",
              },
            ],
            Array [
              "link",
              "image",
              "video",
            ],
            Array [
              Object {
                "align": Array [],
              },
            ],
            Array [
              "clean",
            ],
          ],
        }
      }
      onChange={[Function]}
      placeholder="Add a post to your blog"
      theme="snow"
    />
     
  </div>
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="keywords"
    type="text"
    value=""
  />
   
  <div>
    <button
      className="button"
    >
       Save Post 
    </button>
     
  </div>
   
</form>
`;

exports[`should render PostForm with correct data in 1`] = `
<form
  className="content-container form form--post"
  onSubmit={[Function]}
>
  <UserCard
    author={
      Object {
        "avatar": "http://www.perlworkshop.nl/nlpw2016/img/default-avatar.png",
        "description": "User2 description",
        "email": "",
        "numberOfPosts": 7,
        "stars": 20,
        "topCategories": Array [
          "life",
          "business",
          "history",
        ],
        "username": "user2",
      }
    }
    post={
      Object {
        "author": "user2",
        "authorId": "todo",
        "createdAt": "1970-01-01T00:00:00.000Z",
        "readingTime": 7,
      }
    }
  />
   
  <label
    htmlFor="title"
  >
    <p>
       Title: 
    </p>
     
    <p>
       What should stand for title ? 
    </p>
    <TextArea
      autosize={
        Object {
          "maxRows": 6,
          "minRows": 2,
        }
      }
      onChange={[Function]}
      placeholder="As if you could kill time without injuring eternity."
      prefixCls="ant-input"
      type="text"
      value="Post 1 in Run"
    />
     
  </label>
   
  <div>
    <Quill
      bounds=".app"
      defaultValue={
        Object {
          "ops": Array [
            Object {
              "attributes": "h1",
              "text": "This is the first Post in run",
            },
          ],
        }
      }
      formats={
        Array [
          "header",
          "bold",
          "size",
          "color",
          "background",
          "italic",
          "underline",
          "blockquote",
          "code-block",
          "list",
          "link",
          "image",
          "video",
          "align",
        ]
      }
      modules={
        Object {
          "toolbar": Array [
            Array [
              Object {
                "header": "1",
              },
              Object {
                "header": "2",
              },
            ],
            Array [
              "bold",
              "italic",
              "underline",
              "blockquote",
              "code-block",
            ],
            Array [
              Object {
                "size": Array [
                  "small",
                  false,
                  "large",
                  "huge",
                ],
              },
            ],
            Array [
              Object {
                "color": Array [],
              },
              Object {
                "background": Array [],
              },
            ],
            Array [
              Object {
                "list": "ordered",
              },
            ],
            Array [
              "link",
              "image",
              "video",
            ],
            Array [
              Object {
                "align": Array [],
              },
            ],
            Array [
              "clean",
            ],
          ],
        }
      }
      onChange={[Function]}
      placeholder="Add a post to your blog"
      theme="snow"
    />
     
  </div>
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="keywords"
    type="text"
    value="red, green, blue"
  />
   
  <div>
    <button
      className="button"
    >
       Save Post 
    </button>
     
  </div>
   
</form>
`;
